File: src/handlers/place_trade.rs
pub async fn create_trade_request(
    nats: web::Data<Connection>,
    trade_request: web::Json<PlaceTradeRequest>,
) -> HttpResponse {
    let message_json = match serde_json::to_string(&trade_request.into_inner()) {
        Ok(json) => json,
        Err(e) => {
            eprintln!("Serialization error: {}", e);
            return HttpResponse::InternalServerError().json(json!({
                "status": "error",
                "message": "Failed to serialize trade request"
            }));
        }
    };

    match publish_to_nats(&nats, "place_trade_request", message_json.as_bytes()).await {
        Ok(_) => HttpResponse::Ok().json(json!({
            "status": "success",
            "message": "Trade request published to NATS"
        })),
        Err(e) => {
            eprintln!("NATS publish error: {}", e);
            HttpResponse::InternalServerError().json(json!({
                "status": "error",
                "message": "Failed to publish trade request to NATS"
            }))
        }
    }
}

File: Go/handlers/place_trade.go
package handlers

import (
    "github.com/gin-gonic/gin"
    "github.com/nats-io/nats.go"
    "net/http"
    "stock_broker/models"
    "stock_broker/services"
)

func CreateTradeRequest(nc *nats.Conn) gin.HandlerFunc {
    return func(c *gin.Context) {
        var tradeRequest models.PlaceTradeRequest
        if err := c.ShouldBindJSON(&tradeRequest); err != nil {
            c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request"})
            return
        }

        message, err := json.Marshal(tradeRequest)
        if err != nil {
            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to serialize trade request"})
            return
        }

        err = services.PublishToNATS(nc, "place_trade_request", message)
        if err != nil {
            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to publish trade request to NATS"})
            return
        }

        c.JSON(http.StatusOK, gin.H{"status": "success", "message": "Trade request published to NATS"})
    }
}